File: ruby/algorithms/binary_search.rb
# frozen_string_literal: true

def binary_search(list, item)
  low = 0
  high = list.length - 1

  while low <= high
    mid = low + ((high - low) / 2)
    if  list[mid] == item
      return mid
    elsif list[mid] < item
      low = mid + 1
    else
      high = mid - 1
    end
  end
  "Value not in array"
end

File: javascript/search/binarySearch/binarySearch.js
function binarySearch(arr, item) {
  let low = 0;
  let high = arr.length - 1;

  while (low <= high) {
    const mid = Math.floor(low + (high - low) / 2);

    if (arr[mid] === item) return mid;

    if (arr[mid] < item) low = mid + 1;

    else high = mid - 1;
  }

  return -1;
}

module.exports = binarySearch;
